web-service/helpers.py

```diff
@@ -132,42 +130,48 @@ def SortByDistanceCmp(a, b):
 
 ################################################################################
 
-def FetchAndStoreUrl(siteInfo, url):
+def GetSiteInfoForUrl(url, rssi=None, txpower=None, force_update=False):
+    siteInfo = models.SiteInformation.get_by_id(url)
```

Comment:
Is the cache will be empty for most of the request? It will look up using the shortened URL as a key and the metadata are cached using the expanded URL as a key.